Sources/SwiftNetKit/NetworkService.swift

```diff
@@ -33,4 +51,52 @@ public struct NetworkService {
             throw NetworkError.requestFailed(error: error)
         }
     }
+    
+    func start<Request: RequestProtocol>(_ request: Request, completion: @escaping (Result<Request.ResponseType, Error>) -> Void) {
+        var urlRequest = URLRequest(url: request.url)
+        
+        if let parameters = request.parameters {
+            let queryItems = parameters.map { key, value in
+                URLQueryItem(name: key, value: "\(value)")
+            }
+            var urlComponents = URLComponents(url: request.url, resolvingAgainstBaseURL: false)
+            urlComponents?.queryItems = queryItems
+            urlRequest.url = urlComponents?.url
+        }
+        
+        urlRequest.httpMethod = request.method.stringValue
+        urlRequest.allHTTPHeaderFields = request.headers
+        
+        if let body = request.body {
+            urlRequest.httpBody = body
+        }
+        
+        URLSession.shared.dataTask(with: urlRequest) { data, response, error in
```

Comment:
I would move the above request building into the Request Obj and the Request protocol, since it's specific to each request